test/unit/helper/query_parser.js

```diff
@@ -0,0 +1,168 @@
+
+var parser = require('../../../helper/query_parser');
+var get_layers = require('../../../helper/layers');
+
+module.exports.tests = {};
+
+module.exports.tests.interface = function(test, common) {
+  test('interface', function(t) {
+    t.equal(typeof parser, 'function', 'valid function');
+    t.end();
+  });
+};
+
+module.exports.tests.split_on_comma = function(test, common) {
+  var queries = ['soho, new york', 'chelsea, london', '123 main, new york'];
+  var delim   = ',';
+
+  var testParse = function(query) {
+    test('naive parsing ' + query, function(t) {
+      var address = parser(query);
+      var delimIndex = query.indexOf(delim);
+      var name = query.substring(0, delimIndex);
+      var admin_parts = query.substring(delimIndex + 1).trim();
+
+      t.equal(typeof address, 'object', 'valid object');
+      t.equal(address.name, name, 'name set correctly to ' + address.name);
+      t.equal(address.admin_parts, admin_parts, 'admin_parts set correctly to ' + address.admin_parts);
+      t.end();
+    });
+  };
+
+  for (var keys in queries) {
```

Comment:
`keys` actually has just a single value out of the array, right? so is `key` a better name? It's easy to misread and think keys contains a collection of some kind